File: Exercícios sobre POO/Empresa 8/Empresa em python/funcionario.py
class Funcionario:
    def __init__(self, nome, cargo, salario) -> None:
        self.nome = nome
        self.cargo = cargo
        self.salario = salario
        self.endereco = None

    def adicionar_endereco(self, endereco):
        self.endereco = endereco

    def mostrar_informacoes(self):
        print(f"Nome: {self.nome}, Cargo: {self.cargo}, Salario: {self.salario}")
        if self.endereco:
            self.endereco.mostrar_endereco()
        else:
            print("Endereco não disponivel")

File: Exercícios sobre POO/Empresa 8/Empresa em java/src/Funcionario.java
public class Funcionario {
    private String nome;
    private String cargo;
    private String salario;
    private Endereco endereco;

    public Funcionario(String nome, String cargo, String salario){
        this.nome = nome;
        this.cargo = cargo;
        this.salario = salario;
    }

    public String getNome(){
        return this.nome;
    }

    public void adicionar_endereco(Endereco endereco){
        this.endereco = endereco;
    }

    public void mostrar_informacoes(){
        System.out.println("Nome: " + this.nome + ", Cargo: " + this.cargo + ", Salario: " + this.salario);
        if(this.endereco != null){
            this.endereco.mostrar_endereco();
        }else{
            System.out.println("Endereço não disponivel");
        }
    }
}
